modules/imaging_browser/php/imagingbrowserrowprovisioner.class.inc

```diff
@@ -43,7 +43,7 @@ class ImagingBrowserRowProvisioner extends \LORIS\Data\Provisioners\DBRowProvisi
         // ===========================================================
 
         // Grep all the different scan types present in mri_scan_type
-        $all_scan_types = \Utility::getScanTypeList();
+        $all_scan_types = str_replace('-', '_', \Utility::getScanTypeList());
```

Comment:
There's a Database->quote function for quoting field (or table) names which may have special characters.

Reply:
This was the initial strategy, however I found when using it indiscriminately on all scan types (eg "t1", which didn't have any issues before), it introduced an SQL syntax error for names without any special characters:
> Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''t1pass'

I'll have to implement it with a `preg_match` for special SQL characters.

Reply:
@driusan I just pushed a commit where I added Database->quote calls where needed. However where the names with special characters are used as parts of column names / identifiers, Database->quote won't work because they'll be interpreted as string literals. For those cases I had to use backticks \`  \`. Wouldn't the previous approach (replacing hyphens with underscores) be better since it introduces much less code change?

Reply:
@h-karim The quoting is better practice since you can't always control what users input as fields in a table. the not-so-ideal issue is that we are using field values as column names in another table... that's not great but it's not a problem solvable in this PR.

as far as the quote not working, I think it's better if you terminate the string, add the quote and  re-concatenate the string instead. I made a suggestion

Reply:
`quote` was the wrong function (as per my other comment), it should be `escape` to escape table/column names.